app/views/tasks/index.html.erb

```diff
@@ -0,0 +1,26 @@
+
+<h1>Task List</h1>
+
+<ul class="task-list">
+    <% @tasks.each do|task|%>
+        <li>
+        <%= link_to task.name, task_path(task.id), :class => "task-link"%>
+
+            <section class="index-button">
+
+                <% if task.completed == false %>
+                    <%= link_to "Mark Complete", {action: "mark_complete", id: task.id}, method: :patch %> |
```

Comment:
You should use paths instead of specifying the controller actions like this.

```erb
<%= link_to "Mark Complete", mark_complete_path(task.id) %>
```